compiler/resolution/functionResolution.cpp

```diff
@@ -469,15 +471,19 @@ resolveUninsertedCall(BlockStmt* insideBlock,
   INT_ASSERT(block->parentSymbol);
 
   block->insertAtHead(call);
-  resolveCall(call);
+  if (checkonly && !call->primitive) {
+    resolveNormalCall(call, checkonly);
+  } else {
+    resolveCall(call);
```

Comment:
FYI I've cherry-picked that change into this PR and marked the PR as needing testing. I'm not going to start testing until we're done the review though.